### PR TITLE
Made username and email id unique for each user:PA-615

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
1)Made username and email id unique for each user
2)Removed the delete backup from JustAfterEach in case the backup is deleted as part of the function
**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PA-615

**Special notes for your reviewer**:
log : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test/155/consoleFull
Logs after changes : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test/158/consoleFull

### CODE REVIEW

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -5508,10 +5503,10 @@ var _ = Describe("{ShareAndRemoveBackupLocation}", func() {
 			log.InfoD("Creating %d users", numberOfUsers)
 			var wg sync.WaitGroup
 			for i := 1; i <= numberOfUsers; i++ {
-				userName := fmt.Sprintf("testuser%v", i)
+				userName := fmt.Sprintf("testuser%v-82949", i)
```
Comment (skonda-px) [inline tests/backup/backup_test.go:5506]: you can use createUsers method here.

Comment (snigdha-px) [reply tests/backup/backup_test.go]: done!

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -8166,10 +8145,10 @@ func createUsers(numberOfUsers int) []string {
 	log.InfoD("Creating %d users", numberOfUsers)
 	var wg sync.WaitGroup
 	for i := 1; i <= numberOfUsers; i++ {
-		userName := fmt.Sprintf("testuser%v", i)
+		userName := fmt.Sprintf("testuser%v-%v", i, time.Now().Unix())
 		firstName := fmt.Sprintf("FirstName%v", i)
 		lastName := fmt.Sprintf("LastName%v", i)
-		email := fmt.Sprintf("testuser%v@cnbu.com", i)
+		email := fmt.Sprintf("testuser%v@cnbu.com-%v", i, time.Now().Unix())
```
Comment (skonda-px) [inline tests/backup/backup_test.go:8151]: Since userName is unique, you can use userName@cnbu.com

Comment (snigdha-px) [reply tests/backup/backup_test.go]: done!!

**Review by skonda-px — APPROVED** [review]

**Pull request merged**